### ISSUE TITLE
Logic seems to be reversed

### ISSUE THREAD

**Issue by johnedstone** [protostax/ProtoStax_ISS_Tracker#9]
Thanks for this great project, great example of coding for this application!   I believe that the logic on when to display the `positions` is broken.  Here is a patch, `iss.patch` that fixes this bug.  Thanks again
```
--- iss.py	2022-03-15 20:04:26.258238672 -0400
+++ iss_new.py	2022-03-15 20:06:34.951525600 -0400
@@ -123,6 +123,7 @@
     # Store positions in list
     positions = []
 
+    DISPLAY_REFRESH_INTERVAL_COUNTER = 1
     while(True):
         t0 = time()
         try:
@@ -143,7 +144,7 @@
         print(positions)
 
         # Refresh the display on the first fetch and then on every DISPLAY_REFRESH_INTERVAL fetch
-        if ((len(positions) >= 1) and ((len(positions)-1) % DISPLAY_REFRESH_INTERVAL)):
+        if (len(positions) == 1 ) or DISPLAY_REFRESH_INTERVAL_COUNTER == DISPLAY_REFRESH_INTERVAL: 
             epd.init()
             (imageBlack, imageRed) = display.drawISS(positions)
             # We're drawing the map in black and the ISS location and trajectory in red
@@ -151,6 +152,12 @@
             epd.display(epd.getbuffer(imageBlack), epd.getbuffer(imageRed))
             sleep(2)
             epd.sleep()
+            if (len(positions) > 1):
+                DISPLAY_REFRESH_INTERVAL_COUNTER = 1
+            else:
+                DISPLAY_REFRESH_INTERVAL_COUNTER += 1
+        else:
+            DISPLAY_REFRESH_INTERVAL_COUNTER += 1
        
         t1 = time()
         sleepTime = max(DATA_INTERVAL - (t1 - t0), 0)
```

**Comment by sridharrajagopal** [comment]
Thanks for the catch! This is a 
Duplicate of #8 

The fix is just correcting the condition checking to this (add the == 0 check):

        if ((len(positions) >= 1) and (((len(positions)-1) % DISPLAY_REFRESH_INTERVAL) == 0)):

This has been fixed in the latest commit. Btw, your fix would work as well - achieves the same result, but uses an extra variable that tracks the len(positions). 

Thanks, 
Sridhar

**Comment by johnedstone** [comment]
Perfect!

**Comment by sridharrajagopal** [comment]
Actually, your fix is a better one! Keeping track of DISPLAY_REFRESH_INTERVAL_COUNTER separately from the positions array (as I had done) is better because the positions array is a circular buffer that can fill up, so after it fills up, we'll end up not updating the screen anymore with my logic. I've updated the code with your suggestions (and also tested out the circular buffer filling up logic) - as a bonus, I also added the date/time of the last observation on the bottom of the screen - this will let the user know if the screen has not been updating. 

The changes have been submitted. 

Thanks again!
Sridhar
